x/delegation/keeper/genesis.go

```diff
@@ -66,12 +31,48 @@ func (k Keeper) InitGenesis(
 			panic(errorsmod.Wrap(err, "failed to associate operator with staker"))
 		}
 	}
+
+	// init the state from the general exporting genesis file
+	err := k.SetAllDelegationStates(ctx, gs.DelegationStates)
+	if err != nil {
+		panic(errorsmod.Wrap(err, "failed to set all delegation states"))
+	}
+	err = k.SetAllStakerList(ctx, gs.StakersByOperator)
+	if err != nil {
+		panic(errorsmod.Wrap(err, "failed to set all staker list"))
+	}
+	err = k.SetUndelegationRecords(ctx, gs.Undelegations)
+	if err != nil {
+		panic(errorsmod.Wrap(err, "failed to set all undelegation records"))
+	}
 	return []abci.ValidatorUpdate{}
 }
 
 // ExportGenesis returns the module's exported genesis
-func (Keeper) ExportGenesis(sdk.Context) *delegationtype.GenesisState {
-	genesis := delegationtype.DefaultGenesis()
-	// TODO
-	return genesis
+func (k Keeper) ExportGenesis(ctx sdk.Context) *delegationtype.GenesisState {
+	res := delegationtype.GenesisState{}
+	associations, err := k.GetAllAssociations(ctx)
+	if err != nil {
+		ctx.Logger().Error(errorsmod.Wrap(err, "failed to get all associations").Error())
+	}
+	res.Associations = associations
```

Comment:
directly assign data even though it is failed.

Reply:
I am unsure if an error should influence the export of other fields. I can adjust it to return a null if there is any error.

Reply:
Use panic to address the error according to Max's advice.